extension/src/popup/helpers/blockaid.ts

```diff
@@ -109,3 +114,31 @@ export const useScanTx = () => {
     scanTx,
   };
 };
+
+export const scanAsset = async (
```

Comment:
I moved away from using the hook pattern you were using because I was getting a lot of infinite re-renders when using that pattern with scanAsset. Not sure if you ran into this, too? Looks like you had to disable the exhaustive-deps check on your implementations

Just exporting the `scanAsset` method allowed me to fit it into the existing flows a little more cleanly. These flows already handled loading states, so there wasn't much to gain from using the hook

Reply:
Disabling `exhaustive-deps` is only used to have the effect trigger once on render, so in my cases the deps for the effect triggered re-renders but I only want to run the effect "on mount".

I think this does more or less the same though.